framework/include/base/DependencyResolverInterface.h

```diff
@@ -72,6 +90,9 @@ DependencyResolverInterface::sort(typename std::vector<T> & vector)
 
   for (typename std::vector<T>::iterator iter = start; iter != end ; ++iter)
   {
+    if ((*iter)->allowCyclicDepenedency())
+      continue;
```

Comment:
There is another option, in my previous attempt there was a bunch of new logic added to the dependency interface for "iterative properties", I could restore this stuff and have DiscreteMataial use the iterative stuff.